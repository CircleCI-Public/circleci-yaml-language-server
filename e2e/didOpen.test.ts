import { Commands } from './types';
import {
  configFileUri,
  configFileContent,
  command,
  immediateDiagnostics,
} from './utils';

describe('DidOpen', () => {
  it('Job without steps', async () => {
    const response = await command(
      Commands.DocumentDidOpen,
      {
        textDocument: {
          text: await configFileContent('invalid-files/job-without-steps.yml'),
          uri: configFileUri('invalid-files/job-without-steps.yml'),
          version: 1,
          languageId: 'yaml',
        },
      },
    );

    expect(response).toBeNull();

    const diagnostics = await immediateDiagnostics();

    expect(diagnostics.list).toMatchSnapshot();
  });

  it('Unused job', async () => {
    const response = await command(
      Commands.DocumentDidOpen,
      {
        textDocument: {
          text: await configFileContent('invalid-files/unused-job.yml'),
          uri: configFileUri('invalid-files/unused-job.yml'),
          version: 1,
          languageId: 'yaml',
        },
      },
    );

    expect(response).toBeNull();

    const diagnostics = await immediateDiagnostics();

    expect(diagnostics).toMatchSnapshot();
  });

  it('Detects not existant dockers', async () => {
    const response = await command(
      Commands.DocumentDidOpen,
      {
        textDocument: {
          text: await configFileContent('invalid-files/bad-docker-image.yml'),
          uri: configFileUri('invalid-files/bad-docker-image.yml'),
          version: 1,
          languageId: 'yaml',
        },
      },
    );

    expect(response).toBeNull();

    const diagnostics = await immediateDiagnostics();

    expect(diagnostics).toMatchSnapshot();
  });
});
