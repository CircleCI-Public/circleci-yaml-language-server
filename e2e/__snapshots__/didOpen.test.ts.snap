// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DidOpen Detects not existant dockers 1`] = `
{
  "diagnostics": [],
  "uri": "file:///project/test-files/.circleci/invalid-files/bad-docker-image.yml",
}
`;

exports[`DidOpen Job without steps 1`] = `
{
  "diagnostics": [
    {
      "data": [],
      "message": "Invalid type. Expected: object, given: array",
      "range": {
        "end": {
          "character": 31,
          "line": 8,
        },
        "start": {
          "character": 2,
          "line": 6,
        },
      },
      "severity": 1,
      "source": "cci-language-server",
    },
    {
      "data": [],
      "message": "steps is required",
      "range": {
        "end": {
          "character": 31,
          "line": 5,
        },
        "start": {
          "character": 2,
          "line": 3,
        },
      },
      "severity": 1,
      "source": "cci-language-server",
    },
    {
      "data": [],
      "message": "Job is unused",
      "range": {
        "end": {
          "character": 0,
          "line": 0,
        },
        "start": {
          "character": 0,
          "line": 0,
        },
      },
      "severity": 2,
      "source": "cci-language-server",
    },
  ],
  "uri": "file:///project/test-files/.circleci/invalid-files/job-without-steps.yml",
}
`;

exports[`DidOpen Unused job 1`] = `
{
  "diagnostics": [
    {
      "data": [],
      "message": "Job is unused",
      "range": {
        "end": {
          "character": 6,
          "line": 3,
        },
        "start": {
          "character": 2,
          "line": 3,
        },
      },
      "severity": 2,
      "source": "cci-language-server",
    },
  ],
  "uri": "file:///project/test-files/.circleci/invalid-files/unused-job.yml",
}
`;
